### PR TITLE
Pre-allocate buffers for toString generation

### PR DESCRIPTION
**Pull request by eeSeeGee**

### CODE REVIEW

File: wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java [hunk 1]
```diff
@@ -444,4 +445,24 @@ public void enclosingTypeIsNotMessage() {
...
+
+  @Test
+  public void toStringGeneration() throws IOException {
+    assertThat(new RepoBuilder()
+        .add("message.proto", ""
+            + "message Message {\n"
+            + "  required float long = 1;\n"
+            + "  required string str = 2;\n"
+            + "}\n")
+        .generateCode("Message")).contains("@Override\n"
```
Comment (jrodbx) [inline wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java]: line break before `.contains`

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 2]
```diff
@@ -108,6 +109,12 @@
   static final ClassName NULLABLE = ClassName.get("androidx.annotation", "Nullable");
   static final ClassName CREATOR = ClassName.get("android.os", "Parcelable", "Creator");
 
+  static final String REDACTED_TOSTRING_ASSIGN = "\", $N=██\"";
+  static final int REDACTED_TOSTRING_ASSIGN_LENGTH = REDACTED_TOSTRING_ASSIGN.length() - 2;
+
+  static final String TOSTRING_ASSIGN = "\", $N=\"";
+  static final int TOSTRING_ASSIGN_LENGTH = TOSTRING_ASSIGN.length() - 2;
```
Comment (jrodbx) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: inline the escaped quotes back at the use sites, avoiding the need to offset by 2 here?

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 3]
```diff
@@ -1426,16 +1433,32 @@ private MethodSpec messageHashCode(NameAllocator nameAllocator, MessageType type
     return result.build();
   }
 
-  private MethodSpec messageToString(NameAllocator nameAllocator, MessageType type) {
+  //@VisibleForTesting
```
Comment (jrodbx) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: delete?

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 package com.squareup.wire.java;
 
+import com.google.common.annotations.VisibleForTesting;
```
Comment (jrodbx) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: delete?

**Comment by jrodbx** [comment]
Need to run `./gradlew generateTests`, commit the results and repush

File: wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java (hunk 1 above)
Comment (eeSeeGee) [reply wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java]: Done

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java (hunk 2 above)
Comment (eeSeeGee) [reply wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: It's not the quotes I'm subtracting, it's the `$N`, which I should perhaps add a comment for.

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java (hunk 3 above)
Comment (eeSeeGee) [reply wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: Whoops, yeah

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java (hunk 4 above)
Comment (eeSeeGee) [reply wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: 👍

**Comment by JakeWharton** [comment]
This a fair amount of code. I'm tempted to pre-compute the buffer size as the sum of all names (regardless of whether they're optional) multiplied by 2. That seems like it would prevent approximately 95% of the expansion allocations which seems like a good enough amount given that it only results in a fixed constant flowing into the `StringBuilder` instead of trying to perfectly capture the exact amount.

**Pull request closed without merging**